crates/editor/src/element.rs

```diff
@@ -1558,6 +1589,258 @@ impl EditorElement {
+                        .on_hover({
+                            let editor = self.editor.clone();
+                            let group_id = disp.group_id;
+                            move |hovered, cx| {
+                                editor.update(cx, |editor, cx| {
+                                    let current_group = editor
+                                        .active_diagnostics
+                                        .as_ref()
+                                        .map(|group| group.group_id);
+                                    if *hovered && Some(group_id) != current_group {
+                                        editor.activate_diagnostics(group_id, cx);
+                                    }
+                                });
+                            }
+                        })
+                } else {
+                    div
+                };
+
+                boxes = boxes.child(div);
+            }
+
+            let boxes = if elem.exceeded_max {
+                boxes.child(
+                    h_flex()
+                        .size(line_height)
+                        .h_full()
+                        .w_full()
+                        .bg(Color::Info.color(cx).opacity(0.25))
+                        .child(
+                            Icon::new(IconName::Plus)
+                                .color(Color::Info)
+                                .size(IconSize::Medium),
+                        ),
+                )
+            } else {
+                boxes
+            };
+
+            let mut element = h_flex()
+                .id(SharedString::from(format!("diagnostic-{}", line_ix.0)))
+                .h(line_height)
+                .w_full()
+                .bg(sev_to_color(&elem.severity).color(cx).opacity(0.07))
+                .text_color(sev_to_color(&elem.severity).color(cx))
+                .text_sm()
+                .font_family(style.text.font().family)
+                .child(
+                    h_flex()
+                        .h(line_height)
+                        .gap_2()
+                        .child(boxes)
+                        .child(elem.message),
+                )
+                .into_any();
+
+            element.prepaint_as_root(point(elem.x, elem.y), AvailableSpace::min_size(), cx);
+            elements.insert(line_ix, element);
+        }
+
+        elements
+    }
+
+    #[allow(clippy::too_many_arguments)]
+    fn gather_inline_diagnostics(
+        &self,
+        snapshot: &EditorSnapshot,
+        hitbox: &Hitbox,
+        line_layouts: &[LineWithInvisibles],
+        crease_trailers: &[Option<CreaseTrailerLayout>],
+        content_origin: gpui::Point<Pixels>,
+        scroll_pixel_position: gpui::Point<Pixels>,
+        line_height: Pixels,
+        em_width: Pixels,
+        start_row: DisplayRow,
+        end_row: DisplayRow,
+        cx: &mut WindowContext,
+    ) -> HashMap<DisplayRow, DiagnosticElement> {
```

Comment:
Gathering seems to change between edits, so that different groups are shown, which causes flickering:

https://github.com/user-attachments/assets/de857f54-ecf9-487a-bb44-c8be87694118